sotodlib/preprocess/processes.py

```diff
@@ -107,86 +120,170 @@ class GlitchDetection(_Preprocess):
 
     .. autofunction:: sotodlib.tod_ops.flags.get_glitch_flags
     """
+
     name = "glitches"
-    
+
     def calc_and_save(self, aman, proc_aman):
         glitch_cut, glitch_aman = tod_ops.flags.get_glitch_flags(
-            aman, merge=False, full_output=True,
-            **self.calc_cfgs
-        ) 
+            aman, merge=False, full_output=True, **self.calc_cfgs
+        )
         aman.wrap("glitches", glitch_aman)
         self.save(proc_aman, glitch_aman)
-    
+
     def save(self, proc_aman, glitch_aman):
         if self.save_cfgs is None:
             return
         if self.save_cfgs:
             proc_aman.wrap("glitches", glitch_aman)
- 
+
     def select(self, meta):
         if self.select_cfgs is None:
             return meta
-        
+
         flag = sparse_to_ranges_matrix(
             meta.preprocess.glitches.glitch_detection > self.select_cfgs["sig_glitch"]
         )
         n_cut = count_cuts(flag)
         keep = n_cut <= self.select_cfgs["max_n_glitch"]
         meta.restrict("dets", meta.dets.vals[keep])
         return meta
-    
+
+
+class Jumps(_Preprocess):
+    """Run finding and fixing algorithm.
+    All calculation config go to `find_jumps`
+
+    Saves results in proc_aman under the "jumps" field.
+
+    Data section should define a maximum number of jumps "max_n_jumps".
+
+    .. autofunction:: sotodlib.tod_ops.jumps.find_jumps
+    """
+
+    name = "jumps"
+
+    def process(self, aman, proc_aman):
+        aman.signal = jumpfix_subtract_heights(
+            aman.signal,
+            proc_aman.jumps.jump_flag.mask(),
+            inplace=True,
+            heights=proc_aman.jumps.jump_heights,
+        )
+
+    def calc_and_save(self, aman, proc_aman):
+        jumps, heights = tod_ops.jumps.find_jumps(aman, merge=False, **self.calc_cfgs)
+        jump_aman = tod_ops.jumps.jump_aman(aman, jumps, heights)
+        aman.wrap(name, jump_aman)
```

Comment:
Probably `name` needs to be `self.calc_cfgs['name']`

Reply:
Yeah caught that one this morning